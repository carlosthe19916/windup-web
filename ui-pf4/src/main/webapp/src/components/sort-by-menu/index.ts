export { SortByMenu } from "./sort-by-menu";
