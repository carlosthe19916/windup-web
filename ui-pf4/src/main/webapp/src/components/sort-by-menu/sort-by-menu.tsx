import * as React from "react";
import {
  OverflowMenu,
  OverflowMenuItem,
  Select,
  SelectVariant,
  OptionsMenu,
  OptionsMenuToggle,
  SelectOption,
  SelectOptionObject,
} from "@patternfly/react-core";
import {
  FilterIcon,
  SortAmountUpIcon,
  SortAmountDownIcon,
} from "@patternfly/react-icons";

export interface SortByMenuProps {
  options: Array<string | SelectOptionObject>;
  onChange: (option: string | SelectOptionObject, asc: boolean) => void;
}

export const SortByMenu: React.FC<SortByMenuProps> = ({
  options,
  onChange,
}) => {
  const [isOpen, setIsOpen] = React.useState(false);
  const [selected, setSelected] = React.useState<string | SelectOptionObject>(
    options[0]
  );

  const onToggle = (isExpanded: boolean) => {
    setIsOpen(isExpanded);
  };

  const onSelect = (
    _: React.MouseEvent | React.ChangeEvent,
    value: string | SelectOptionObject
  ) => {
    setSelected(value);
    setIsOpen(false);

    onChange(selected, false);
  };

  return (
    <OverflowMenu breakpoint="md">
      <OverflowMenuItem isPersistent>
        <Select
          variant={SelectVariant.single}
          aria-label="Select sort by"
          aria-labelledby="page-layout-table-column-management-action-toolbar-top-select-checkbox-label page-layout-table-column-management-action-toolbar-top-select-checkbox-toggle"
          placeholderText={
            <>
              <FilterIcon /> {selected}
            </>
          }
          onToggle={onToggle}
          onSelect={onSelect}
          selections={selected}
          isOpen={isOpen}
        >
          {options.map((el, index) => (
            <SelectOption key={index} value={el} />
          ))}
        </Select>
      </OverflowMenuItem>
      <OverflowMenuItem>
        <OptionsMenu
          id="page-layout-table-column-management-action-toolbar-top-options-menu-toggle"
          isPlain
          menuItems={[]}
          toggle={
            <OptionsMenuToggle
              toggleTemplate={<SortAmountDownIcon aria-hidden="true" />}
              aria-label="Sort by"
              // onToggle={(a: boolean) => console.log(a)}
              onClick={() => console.log("a")}
              hideCaret
            />
          }
        />
      </OverflowMenuItem>
    </OverflowMenu>
  );
};
