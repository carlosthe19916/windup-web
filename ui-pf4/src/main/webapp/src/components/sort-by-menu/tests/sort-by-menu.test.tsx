import * as React from "react";
import { shallow } from "enzyme";
import { SortByMenu } from "../sort-by-menu";

describe("Welcome", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(
      <SortByMenu options={["Option1", "Option2"]} onChange={jest.fn} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
