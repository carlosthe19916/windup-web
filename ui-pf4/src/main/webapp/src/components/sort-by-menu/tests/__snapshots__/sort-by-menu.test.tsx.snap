// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Renders without crashing 1`] = `
<OverflowMenu
  breakpoint="md"
>
  <OverflowMenuItem
    isPersistent={true}
  >
    <Select
      aria-label="Select sort by"
      aria-labelledby="page-layout-table-column-management-action-toolbar-top-select-checkbox-label page-layout-table-column-management-action-toolbar-top-select-checkbox-toggle"
      className=""
      clearSelectionsAriaLabel="Clear all"
      createText="Create"
      customBadgeText={null}
      customContent={null}
      direction="down"
      hasInlineFilter={false}
      inlineFilterPlaceholderText={null}
      inputIdPrefix=""
      isCreatable={false}
      isDisabled={false}
      isGrouped={false}
      isOpen={false}
      isPlain={false}
      menuAppendTo="inline"
      noResultsFoundText="No results found"
      onClear={[Function]}
      onCreateOption={[Function]}
      onFilter={null}
      onSelect={[Function]}
      onToggle={[Function]}
      ouiaSafe={true}
      placeholderText={
        <React.Fragment>
          <FilterIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
           
          Option1
        </React.Fragment>
      }
      removeSelectionAriaLabel="Remove"
      selections="Option1"
      toggleAriaLabel="Options menu"
      toggleIcon={null}
      toggleId={null}
      typeAheadAriaLabel=""
      variant="single"
      width=""
    >
      <SelectOption
        className=""
        component="button"
        index={0}
        inputId=""
        isChecked={false}
        isDisabled={false}
        isNoResultsOption={false}
        isPlaceholder={false}
        isSelected={false}
        key="0"
        keyHandler={[Function]}
        onClick={[Function]}
        sendRef={[Function]}
        value="Option1"
      />
      <SelectOption
        className=""
        component="button"
        index={0}
        inputId=""
        isChecked={false}
        isDisabled={false}
        isNoResultsOption={false}
        isPlaceholder={false}
        isSelected={false}
        key="1"
        keyHandler={[Function]}
        onClick={[Function]}
        sendRef={[Function]}
        value="Option2"
      />
    </Select>
  </OverflowMenuItem>
  <OverflowMenuItem>
    <OptionsMenu
      id="page-layout-table-column-management-action-toolbar-top-options-menu-toggle"
      isPlain={true}
      menuItems={Array []}
      toggle={
        <OptionsMenuToggle
          aria-label="Sort by"
          hideCaret={true}
          onClick={[Function]}
          toggleTemplate={
            <SortAmountDownIcon
              aria-hidden="true"
              color="currentColor"
              noVerticalAlign={false}
              size="sm"
            />
          }
        />
      }
    />
  </OverflowMenuItem>
</OverflowMenu>
`;
